core/src/avm2/array.rs

```diff
@@ -21,8 +214,10 @@ impl<'gc> ArrayStorage<'gc> {
     /// The length parameter indicates how big the array storage should start
     /// out as. All array storage consists of holes.
     pub fn new(length: usize) -> Self {
-        Self {
-            storage: vec![None; length],
+        if length > MAX_DENSE_LENGTH {
+            ArrayStorage::Sparse(BTreeMap::new(), 0)
```

Comment:
Why does the inner ArrayStorage take a length of 0 if we have a length parameter?